File: Human_written_code/Code_with_comments/Q3.cpp
#include <bits/stdc++.h>
using namespace std;

// Sort an array of 0s, 1s, and 2s
void sortArray(vector<int>& arr, int n) {
    int low = 0, mid = 0, high = n - 1;

    while (mid <= high) {
        if (arr[mid] == 0) {
            swap(arr[low], arr[mid]); // Move 0s to the beginning
            low++;
            mid++;
        }
        else if (arr[mid] == 1) {
            mid++; // Skip 1s as they're already in place
        }
        else {
            swap(arr[mid], arr[high]); // Move 2s to the end
            high--;
        }
    }
}

int main() {
    vector<int> arr = {0, 2, 1, 2, 0, 1};
    sortArray(arr, arr.size());

    cout << "After sorting:" << endl;
    for (int i = 0; i < arr.size(); i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level1/Q3.py
def sort_colors(nums):
    # Initialize pointers for the current element, 0 (low), and 2 (high)
    low, mid, high = 0, 0, len(nums) - 1
    
    while mid <= high:
        if nums[mid] == 0:
            # Swap the mid element with the low element
            nums[low], nums[mid] = nums[mid], nums[low]
            low += 1
            mid += 1
        elif nums[mid] == 1:
            # Move the mid pointer up
            mid += 1
        else:
            # Swap the mid element with the high element
            nums[high], nums[mid] = nums[mid], nums[high]
            high -= 1
    
    return nums

